tests/testthat/test-decision_tree-rpart.R

```diff
@@ -95,6 +95,19 @@ test_that("survival predictions", {
   expect_equal(f_pred$.pred[[1]]$.eval_time, 100:200)
 })
 
+test_that("can predict for out-of-domain timepoints", {
+  eval_time_obs_max_and_ood <- c(1022, 2000)
+  obs_without_NA <- lung[2,]
+
+  mod <- decision_tree() %>%
+    set_mode("censored regression") %>%
+    set_engine("rpart") %>%
+    fit(Surv(time, status) ~ ., data = lung)
+
+  expect_no_error(
+    preds <- predict(mod, obs_without_NA, type = "survival", eval_time = eval_time_obs_max_and_ood)
```

Comment:
This prediction differs in output format than then other ones with this one being

```r
str(preds)
tibble [1 × 1] (S3: tbl_df/tbl/data.frame)
 $ .pred: list<tibble[,2]> [1:1] 
  ..$ : tibble [2 × 2] (S3: tbl_df/tbl/data.frame)
  .. ..$ .eval_time    : num [1:2] 1022 2000
  .. ..$ .pred_survival: num [1:2] NA NA
  ..@ ptype: tibble [0 × 2] (S3: tbl_df/tbl/data.frame)
  .. ..$ .eval_time    : num(0) 
  .. ..$ .pred_survival: num(0) 
```

opossed to partykit which produces

```r
str(preds)
tibble [1 × 1] (S3: tbl_df/tbl/data.frame)
 $ .pred:List of 1
  ..$ : tibble [2 × 2] (S3: tbl_df/tbl/data.frame)
  .. ..$ .eval_time    : num [1:2] 1022 2000
  .. ..$ .pred_survival: num [1:2] 0.0499 0.0499
```